tests_v2/cmux.py

```
@@ -54,29 +74,77 @@ def _read_last_socket_path() -> Optional[str]:
     return None
 
 
+def _can_connect(path: str, timeout: float = 0.15, retries: int = 4) -> bool:
+    # Best-effort check to avoid getting stuck on stale socket files.
+    for _ in range(max(1, retries)):
+        s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        try:
+            s.settimeout(timeout)
+            s.connect(path)
+            return True
```

Ruff (0.15.12)

[warning] Consider moving this statement to an `else` block

(TRY300)